### ISSUE TITLE
RequiredIf attribute prevents DisplayName from working

### ISSUE THREAD

**Issue by im1dermike** [jwaliszko/ExpressiveAnnotations#94]
I just noticed that when I have the `[RequiredIf]` attribute on my view model properties, the display name value isn't being used for labels.  Eg:

View model:

    public class RegistrationViewModel
    {
        [RequiredIf("IsAdmin == false", ErrorMessage = "Date of Birth is required")]
        [Display(Name = "Date of Birth")]
        public DateTime? DateOfBirth { get; set; }
    }

View:

    @Html.LabelFor(m => m.DateOfBirth)

Instead of "Date of Birth" being displayed, it's displaying "DateOfBirth".  The way only to get it to use the display name now is to include it in the Razor code like this:

    @Html.LabelFor(m => m.DateOfBirth, "Date of Birth")

I would prefer to keep it in the view model.

When I comment out the `[RequiredIf]` attribute, it uses the display name correctly.

Unfortunately, I can't test in the ExpressiveAnnotations samples because they're in an newer version of .NET.

**Comment by jwaliszko** [comment]
It's surprising and looks like a problem related to `LabelFor` helper, not EA. Does the `LableFor` helper you're using cannot extract display name when such a field has multiple annotations attached to it? It would be quite unexpected. What ASP.NET MVC version are you using (EA has been tested on MVC4 and 5)? Is this `LabelFor` your custom extension or framework one?

**Comment by im1dermike** [comment]
I actually just noticed implementing the client-side validation (which I did this morning) broke all `[RequiredIf]` functionality, including the display name.

I referenced ExpressiveAnnotations.dll and ExpressiveAnnotations.MvcUnobtrusive.dll in my web app.

I added `DataAnnotationsModelValidatorProvider.RegisterAdapter(typeof(RequiredIfAttribute), typeof(RequiredIfValidator));` to my Global.asax Application_Start().  

And I included `expressive.annotations.validate.js` in a bundle after `jquery.validate.js` and `jquery.validate.unobtrusive.js`.

What did I do wrong?

**Comment by im1dermike** [comment]
Actually commenting all of that out and simply including the ExpressiveAnnotation DLLs in the web app project break `[RequiredIf]` for me.

I'm using .NET 4.5.

**Comment by jwaliszko** [comment]
What ASP.NET MVC version? 
I assume that the versions of dependencies on which EA relays are correct, i.e.
* Newtonsoft.Json (≥ 4.5.11)
* jQuery (≥ 1.8.2)
* jQuery.Validation (≥ 1.10.0)
* Microsoft.jQuery.Unobtrusive.Validation (≥ 3.1.1)

**Comment by im1dermike** [comment]
Here is what I have:

System.Web.Mvc.dll: 5.2.3.0
Newtonsoft.Json.dll: 7.0.0

jquery-2.1.4.js
jquery.validate.js: 1.14.0
jquery.validate.unobtrusive.js: Inexplicably doesn't give the version
jquery.unobtrusive-ajax.js: Inexplicably doesn't give the version

**Comment by jwaliszko** [comment]
Are you able to provide me explicit and tiny project (zip archive) where this single error is reproducible?

**Comment by im1dermike** [comment]
I'll see if I can create a new tiny project to reproduce sometime soon.

**Comment by jwaliszko** [comment]
Due to the fact I'm unable to reproduce this issue, I'm closing it. If you find spare time to get back with required details in the future, reopen it please.
